paymentsheet/src/main/java/com/stripe/android/paymentsheet/verticalmode/PaymentMethodVerticalLayoutInteractor.kt

```diff
@@ -206,8 +206,10 @@ internal class DefaultPaymentMethodVerticalLayoutInteractor(
 
     init {
         coroutineScope.launch {
-            // The first screen change is navigating to us! We don't want to clear in that case.
-            isCurrentScreen.drop(1).collect { isCurrentScreen ->
+            // We expect this to be false, true, then whatever changes actually happen.
+            // The first screen is likely loading (before it gets to us), then the navigation happens to us.
+            // We're dropping the initial 2, so we don't get events until after we're on the vertical mode screen.
+            isCurrentScreen.drop(2).collect { isCurrentScreen ->
```

Comment:
I'm a bit surprised that this update didn't cause the tests added/modified in https://github.com/stripe/stripe-android/pull/9201 to fail. Do we need to update them to ensure they're still testing the right thing?

Reply:
I did update one of them... But it's kinda hard. It's a bit of a test vs production setup thing. 

But to be honest, yeah. I'm not super happy about it.

I'm planning to add an integration test soon though. Which will make this difference between test and production setup have the integration coverage we really need here.

Reply:
ok, yeah an integration test for this sounds great!